packages/cli/src/lib/index.js

```diff
@@ -190,6 +195,29 @@ export const prettierOptions = () => {
   }
 }
 
+// TODO: Move this into `generateTemplate` when all templates have TS support
+/*
+ * Convert a generated TS template file into JS.
+ */
+export const transformTSToJS = (filename, content) => {
+  const result = babel.transform(content, {
+    filename,
+    configFile: false,
+    plugins: [
+      [
+        '@babel/plugin-transform-typescript',
+        {
+          isTSX: true,
+          allExtensions: true,
+        },
+      ],
+    ],
+    retainLines: true,
+  }).code
+
+  return prettify(filename.replace(/\.ts$/, '.js'), result)
```

Comment:
Now that we have .tsx, does all the extension magic still work?

Reply:
I pass the final `.js` output filename to `transformTSToJS` [here](https://github.com/redwoodjs/redwood/pull/632/files#diff-84db5e5197e6a75e3d30be8c5e732bfaR34), so teeeechnically the PR doesn't use this `replace` logic. I can remove it, and it can be re-added whenever we end up needing it. What do you think?

Reply:
I'd vote to remove. But defer to your decision.

Reply:
It's in master now, so I'll amend it to cover the `.tsx` case in case other generators end up needing it too.